spec/re_captcha/configuration_spec.rb

```diff
@@ -26,20 +26,86 @@
     end
 
     describe '#valid?' do
+      let(:errors) { double('Array', :<< => nil) }
+
+      it 'resets errors' do
+        expect(instance).to receive(:reset_errors)
+        instance.valid?
+      end
+
+      it 'sets errors' do
+        expect(instance).to receive(:set_errors)
```

Comment:
+1 the issue with setting up expectations on private methods is that private implementation can change during refactoring phase. But usually the public interface stays the same, its behavior in particular. It's a pain to reflect private changes to the specs as well whereas if you tested the expected behavior your tests will still pass and can actually help you to perform confident refactoring.